### PR TITLE
add dataset counts script

### PR DESCRIPTION
**Pull request by CloseChoice**
This can be used for a part of #2174.

This PR contains:
 - a generic script to count the dataset entries for any dataset and any mode
 - updated model/model_training/README.md with some counts of datasets
 - some minor adjustments to model/model_training/utils.py so that the datasets can be returned as list. This is done so that this PR can reuse existing code and does not need to copy+paste.

### CODE REVIEW

File: model/model_training/configs/config.yaml [hunk 1]
```diff
@@ -29,7 +29,7 @@ defaults:
     - cnn_dailymail
     - multi_news
     - scitldr
-    - soda
+    # - soda  # TODO: fix list index out of range error
```
Comment (dvruette) [inline model/model_training/configs/config.yaml:32]: still to do?

**Review by dvruette — APPROVED** [review]
LGTM, thanks!

File: model/model_training/configs/config.yaml (hunk 1 above)
Comment (CloseChoice) [reply model/model_training/configs/config.yaml:32]: that's still an open todo, just checked it. Will create a seperate PR to fix this.

File: model/model_training/utils.py [hunk 2]
```diff
@@ -365,6 +367,8 @@ def get_dataset(conf, mode="sft"):
         if val is not None:
             evals[dataset_name] = Subset(val, list(range(min(len(val), conf.eval_size)))) if conf.eval_size else val
 
+    if return_datasets_as_list:
```
[the remark this answers is not in the document]
Comment (CloseChoice) [reply model/model_training/utils.py]: Thanks for pointing me towards this. Just removed the `return_datasets_as_list` part ;)

**Pull request merged**